Makefile

```diff
@@ -119,8 +119,14 @@ oldtest: | nvim build/runtime/doc/tags
 ifeq ($(strip $(TEST_FILE)),)
 	+$(SINGLE_MAKE) -C src/nvim/testdir NVIM_PRG="$(realpath build/bin/nvim)" $(MAKEOVERRIDES)
 else
-	+$(SINGLE_MAKE) -C src/nvim/testdir NVIM_PRG="$(realpath build/bin/nvim)" SCRIPTS= $(MAKEOVERRIDES) $(TEST_FILE)
+	@# Handle TEST_FILE=test_foo{,.res,.vim}.
+	+$(SINGLE_MAKE) -C src/nvim/testdir NVIM_PRG="$(realpath build/bin/nvim)" \
+	  $(patsubst %.vim,%,$(patsubst %.res,%,$(TEST_FILE))) SCRIPTS= $(MAKEOVERRIDES)
 endif
+# Build oldtest by specifying the .vim filename.
+.PHONY: phony_force
+src/nvim/testdir/%.vim: phony_force
+	$(MAKE) oldtest SCRIPTS= NEW_TESTS_RES=$(patsubst src/nvim/testdir/%.vim,%.res,$@)
```

Comment:
Specifying the full `src/nvim/testdir/%.vim` path is not worth all this complication. If there's a way we can support this:

    TEST_FILE=foo.vim make oldtest

then that's good enough, and strongly preferred if it avoids a bunch of extra Makefile tricks.

Reply:
I think we can support both, and thought that the full path is better, since it allows for tab completion of filenames, and also goes into the direction of the `test/run.sh` script I suggested somewhere, and where you've said it would be better to have it in / via the Makefile.

Reply:
> also goes into the direction of the `test/run.sh` script I suggested somewhere, and where you've said it would be better to have it in / via the Makefile.

`test/run.sh` is yet more indirection... definitely better to keep things in the Makefile, if it's something that's really needed at all. But that's also a compromise because CMake is the build system, not make...

I also am uncomfortable with gold-plating "oldtest" in general. `TEST_FILE=foo.vim` is good to have for parallel form with the Lua tests. That's the only reason.

Reply:
> But that's also a compromise because CMake is the build system, not make...

Not for oldtests though.  Here our main Makefile redirects to the local Makefile there.

> I also am uncomfortable with gold-plating "oldtest" in general. `TEST_FILE=foo.vim` is good to have for parallel form with the Lua tests. That's the only reason.

It's only done there first.  The same pattern would also work then for other tests in general.

Reply:
> Not for oldtests though. Here our main Makefile redirects to the local Makefile there.

That's only because no one converted it to a proper CMake target. (Though we did so with many other Vim Makefiles, including docs IIRC.)